kernel/kernel-api/src/main/java/io/delta/kernel/internal/metrics/SnapshotQueryContext.java

```diff
@@ -75,7 +84,7 @@ public static SnapshotQueryContext forTimestampSnapshot(String tablePath, long t
    * @param providedTimestamp the provided timestamp for a time-travel-by-timestamp query, empty if
    *     this is not a time-travel-by-timestamp query
    */
-  private SnapshotQueryContext(
+  protected SnapshotQueryContext(
```

Comment:
@allisonport-db I think we might want to refactor this to be more extensible FYI